src/sdk/pynni/nni/compression/torch/utils/mask_conflict.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import logging
+import torch
+import numpy as np
+from .shape_dependency import ChannelDependency
+# logging.basicConfig(level = logging.DEBUG)
+_logger = logging.getLogger('FixMaskConflict')
+
+class MaskConflict:
+    def __init__(self, mask_file, model=None, dummy_input=None, graph=None):
+        """
+        MaskConflict fix the mask conflict between the layers that
+        has channel dependecy with each other.
+
+        Parameters
+        ----------
+        model : torch.nn.Module
+            model to fix the mask conflict
+        dummy_input : torch.Tensor
+            input example to trace the model
+        mask_file : str
+            the path of the original mask file
+        graph : torch._C.Graph
+            the traced graph of the target model, is this parameter is not None,
+            we donnot use the model and dummpy_input to get the trace graph.
+        """
+        # check if the parameters are valid
+        parameter_valid = False
+        if graph is not None:
+            parameter_valid = True
+        elif (model is not None) and (dummy_input is not None):
+            parameter_valid = True
+        if not parameter_valid:
+            raise Exception('The input parameters is invalid!')
+        self.model = model
+        self.dummy_input = dummy_input
+        self.graph = graph
+        self.mask_file = mask_file
+        self.masks = torch.load(self.mask_file)
+
+    def fix_mask_conflict(self):
+        """
+        Fix the mask conflict before the mask inference for the layers that
+        has shape dependencies. This function should be called before the
+        mask inference of the 'speedup' module.
+        """
+        channel_depen = ChannelDependency(self.model, self.dummy_input, self.graph)
+        depen_sets = channel_depen.dependency_sets
+        for dset in depen_sets:
+            if len(dset) == 1:
+                # This layer has no channel dependency with other layers
+                continue
+            channel_remain = set()
+            fine_grained = False
+            for name in dset:
+                if name not in self.masks:
+                    # this layer is not pruned
+                    continue
+                w_mask = self.masks[name]['weight']
+                shape = w_mask.size()
+                count = np.prod(shape[1:])
+                all_ones = []
+                all_zeros = []
+                for i in range(w_mask.size(0)):
+                    _count = torch.sum(w_mask[i])
+                    if _count == count:
+                        all_ones.append(i)
+                    elif _count == 0:
+                        all_zeros.append(i)
```

Comment:
better use matrix op to replace for loop if possible for performance, such as:
`channel_pruned = (w_mask.flatten(1).sum(-1) >= 1).sum().item() < w_mask.size(0)`